src/main/kotlin/com/openlattice/mechanic/upgrades/V3TimeUseDiaryUpgrade.kt

```diff
@@ -0,0 +1,437 @@
+ * entity data model. Inserts submissions into a postgres table
+ * for migration to Chronicle v3
+ *
+ */
+class V3TimeUseDiaryUpgrade(
+    private val toolbox: Toolbox,
+    private val hds: HikariDataSource,
+    private val pgEntityDataQueryService: PostgresEntityDataQueryService,
+    private val searchService: SearchService,
+    private val principalService: HazelcastPrincipalService,
+    ) : Upgrade {
+
+    private val propertyTypes = HazelcastMap.PROPERTY_TYPES.getMap(toolbox.hazelcast)
+    private val entitySets = HazelcastMap.ENTITY_SETS.getMap(toolbox.hazelcast)
+    private val entitySetIds = HazelcastMap.ENTITY_SETS.getMap(toolbox.hazelcast).values.associateBy { it.name }
+    private val organizations = HazelcastMap.ORGANIZATIONS.getMap(toolbox.hazelcast)
+    private val answerPropertyTypeIds = propertyTypes.getAll(setOf(OL_VALUES_ID, OL_ID_ID))
+    // TODO: replace empty strings with chronicle super user ids (auth0 and google-oauth2) when running migration
+    private val chronicleSuperUserIds = setOf("")
+
+    companion object {
+
+        val logger = LoggerFactory.getLogger(V3TimeUseDiaryUpgrade::class.java)
+        val objectMapper = ObjectMapper().registerModule( JavaTimeModule() )
+            .registerModule(SimpleModule().addSerializer(OffsetDateTime::class.java, OffsetDateTimeSerializer<OffsetDateTime>()))
+
+        private val COMPLETED_DT_FQN = FullQualifiedName("date.completeddatetime")
+        private val SUBJECT_ID_FQN = FullQualifiedName("nc.SubjectIdentification")
+        private val OL_TITLE_FQN = FullQualifiedName("ol.title")
+        private val OL_ID_FQN = FullQualifiedName("ol.id")
+        private val OL_DT_START_FQN = FullQualifiedName("ol.datetimestart")
+        private val OL_DT_END_FQN = FullQualifiedName("ol.datetimeend")
+        private val OL_VALUES_FQN = FullQualifiedName("ol.values")
+        private val OL_VALUES_ID = UUID.fromString("dcc3bc24-3a5d-45cf-8e38-bc9ba8c43d06")
+        private val OL_ID_ID = UUID.fromString("39e13db7-a730-421a-a600-ae0674060140")
+        private val OL_STUDY_ID = UUID.fromString("80c86a96-0e3f-46eb-9fbb-60d9174566a5")
+        private val answerEntityTypeId = UUID.fromString("7912f235-1959-4dd2-8a61-d85cd09b0c34")
+
+        private val dropExistingTimeUseDiaryTableSql = """
+            DROP TABLE IF EXISTS public.time_use_diary_submissions;
+        """.trimIndent()
+
+        private val createTimeUseDiarySubmissionsTableSql = """        
+            CREATE TABLE public.time_use_diary_submissions (
+                submission_id uuid NOT NULL,
+                organization_id uuid NOT NULL,
+                study_id uuid NOT NULL,
+                participant_id text NOT NULL,
+                submission_date timestamp with time zone DEFAULT now() NOT NULL,
+                submission jsonb
+            );
+        """.trimIndent()
+
+        private val addPrimaryKeyToTimeUseDiaryTableSql = """
```

Comment:
Couldn't you just specify submission as primary key in CREATE TABLE?

Reply:
Ah, when I pg_dump-ed the v3 schema it came out as two separate commands but I overlooked combining them. Will update to be a single command.